data/binding/json.go

```diff
@@ -0,0 +1,327 @@
+package binding
+
+import (
+	"errors"
+	"sync"
+
+	"fyne.io/fyne/v2/data/binding"
+
+	jsonvalue "github.com/Andrew-M-C/go.jsonvalue"
+)
+
+// JSONValue supports binding a jsonvalue.V{}
+type JSONValue interface {
+	binding.DataItem
+
+	GetItemString(firstParam interface{}, params ...interface{}) (binding.String, error)
+	GetItemFloat(firstParam interface{}, params ...interface{}) (binding.Float, error)
+	GetItemBool(firstParam interface{}, params ...interface{}) (binding.Bool, error)
+
+	IsEmpty() bool
+}
+
+type databoundJSON struct {
+	JSONValue
+
+	self   binding.Untyped
+	rwlock sync.RWMutex
+	source binding.String
+	last   string
+	err    error
+}
+
+// Internal type for the children accessor
+type childJSON struct {
+	source JSONValue
```

Comment:
If this was of type `*databoundJSON` you could lose a lot of casts required for the `get()` and `set(...)` calls.